bash_completion

```diff
@@ -3298,45 +3303,65 @@ else
     }
 fi
 
-# source compat completion directory definitions
-_comp__init_compat_dirs=()
-if [[ ${BASH_COMPLETION_COMPAT_DIR-} ]]; then
-    _comp__init_compat_dirs+=("$BASH_COMPLETION_COMPAT_DIR")
-else
-    _comp__init_compat_dirs+=(/etc/bash_completion.d)
-    # Similarly as for the "completions" dir, look up from relative to
-    # bash_completion, primarily for installed-with-prefix and
-    # run-in-place-from-git-clone setups.  Notably we do it after the system
-    # location here, in order to prefer in-tree variables and functions.
-    if [[ ${BASH_SOURCE%/*} == */share/bash-completion ]]; then
-        _comp__init_compat_dir=${BASH_SOURCE%/share/bash-completion/*}/etc/bash_completion.d
-    elif [[ $BASH_SOURCE == */* ]]; then
-        _comp__init_compat_dir="${BASH_SOURCE%/*}/bash_completion.d"
+# List custom/extra completion files to source on the startup
+## @param $1 path Path to "bash_completion"
+## @var[out] _comp__init_startup_configs
+_comp__init_collect_startup_configs()
```

Comment:
I suppose we could `unset -f` this as it is only being used at load time?

Pre-approved with that.

Reply:
Yes, and it already unsets itself at the end.

https://github.com/scop/bash-completion/pull/1212/files#diff-a4757074ff650000804fd3eaabe9b0a9e02e33040ca5b8afd4c0275fc5f3e136R3357

Is it better to do it at the beginning of the function?

Reply:
Ah! Missed that.

I don't know if it matters much where the `unset` happens, if something is only to be called once anyway. Perhaps at the end sounds like the best option for me. We are not consistent with that currently (could address in a followup MR if we care):
- `_comp_dequote__initialize`: beginning
- `_comp__init_set_up_service_completions`, `_comp_cmd_tar`, `_comp_cmd_vncviewer__bootstrap`: end
- `_comp_cmd_makepkg__bootstrap`: middle

Reply:
I'll submit a PR.

As for `_comp_cmd_makepkg__bootstrap`, I guess it wanted to put `$func "$@"` at the end of the function to reflect the exit status.

Reply:
#1215